### ISSUE TITLE
Accessibility issues with Grid and sub controls like Excel Style Filtering

### ISSUE THREAD

**Issue by ericmahlberg** [IgniteUI/igniteui-angular#9028]
## Description  
There are several accessibility issues pertaining to the igx-Grid and the Excel Style Filtering dialogue.

 * igniteui-angular version: as early as 10.1.9 through to the latest 11.x
 * browser: Chrome

## Steps to reproduce  
1. Navigate to https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering
2. Open one of the column filter dialogues so that the excel like options are visible
3. Run an accessibility tool against the page (https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni)

## Result  
Many violations flagged for accessibility within the grid itself as well as the Excel filter dialogues
Two examples:
***************
Title: WCAG 4.1.2: Ensures ARIA attributes are allowed for an element's role (.igx-grid__thead-wrapper)
Tags: Accessibility, WCAG 4.1.2, aria-allowed-attr
Issue: Ensures ARIA attributes are allowed for an element's role (aria-allowed-attr - https://accessibilityinsights.io/info-examples/web/aria-allowed-attr)
Element path: .igx-grid__thead-wrapper
Snippet: <div role="rowgroup" tabindex="0" class="igx-grid__thead-wrapper" aria-activedescendant="statusTrackerGrid_undefined" style="width: 1785px;">
How to fix: 
Fix any of the following:
 ARIA attribute is not allowed: aria-activedescendant="statusTrackerGrid_undefined"
***************
Title: WCAG 4.1.2: Ensures buttons have discernible text (.igx-button--icon[igxbutton="icon"][ng-reflect-type="icon"]:nth-child(2))
Tags: Accessibility, WCAG 4.1.2, button-name
Issue: Ensures buttons have discernible text (button-name - https://accessibilityinsights.io/info-examples/web/button-name)
Element path: .igx-excel-filter__menu-header-actions > .igx-button--icon[igxbutton="icon"][ng-reflect-type="icon"]:nth-child(2)
Snippet: <button igxbutton="icon" class="igx-button--icon ng-star-inserted" ng-reflect-type="icon" role="button">
How to fix: 
Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element's default semantics were not overridden with role="none" or role="presentation"
  Element has no title attribute

## Expected result  
No violations flagged for accessibility

## Attachments  
Attach a sample if available, and screenshots, if applicable.  
![Screenshot 2021-02-19 122544](https://user-images.githubusercontent.com/7574313/108557577-a6e70a00-72ad-11eb-9eff-23a47e38cf22.png)

**Comment by zdrawku** [comment]
Hello @ericmahlberg, thank you for pointing that out! If we ignore the color-contrast issues (as they are logged in [separate issue](https://github.com/IgniteUI/igniteui-angular/issues/7160)) the only problem is that *Aria attribute is not allowed*, part of the thead wrapper element - rowgroup? Same applies to the excel-style filtering errors.

As a temporary solution for the contrast issues you can follow [this suggestion](https://github.com/IgniteUI/igniteui-angular/issues/7037#issuecomment-614732588). As for the `aria-activedescendant` attribute, we are going to investigate the error.

**Comment by ericmahlberg** [comment]
Thanks @zdrawku - I've addressed the color contrast issues via CSS in our instance.  

Regarding the aria-activedescendant issue, I see it not only in the thead-wrapper, but the tbody-content and tfoot elements as well.

**Comment by ericmahlberg** [comment]
Thanks @zdrawku and team!
Will you be incorporating this into a release soon?  I checked releases and didn't see the commit there yet.  Also, it would be great if this could be included in a 10x release as we haven't hopped to 11 yet.

**Comment by zdrawku** [comment]
hi @ericmahlberg, yes, @Aleksandyr will add this fix to the latest version of the branches that are still maintained - 10.2.x and 11.1.x

**Comment by ericmahlberg** [comment]
Hi @zdrawku, @Aleksandyr  - I saw that this commit was incorporated into 10.2.19 so I pulled it down. 

I still see 2 issues regarding the icon buttons in the top right hand of the excel style filter dialog.

![image](https://user-images.githubusercontent.com/7574313/113610813-7ad7dc00-9602-11eb-8225-27a6781d1ee6.png)

![image](https://user-images.githubusercontent.com/7574313/113610715-5845c300-9602-11eb-800e-8f0f92bed3a0.png)

**Comment by zdrawku** [comment]
hey @ericmahlberg, this is not exactly part of the contex of this issue. We just discussed with @DiyanDimitrov, that we might need to review all FIltering containers (excel-style filtering, advanced filtering, default filtering) and with that said could you please add this as a separate issue.

**Comment by ericmahlberg** [comment]
Done - link: https://github.com/IgniteUI/igniteui-angular/issues/9287
